nilearn/mass_univariate/permuted_least_squares.py

```diff
@@ -663,359 +968,464 @@ def permuted_ols(
         )
 
-    target_vars = np.asfortranarray(target_vars)  # efficient for chunking
-    n_descriptors = target_vars.shape[1]
-    if np.any(np.all(target_vars == 0, axis=0)):
-        warnings.warn(
-            "Some descriptors in 'target_vars' have zeros across all samples. "
-            "These descriptors will be ignored during null distribution "
-            "generation."
-        )
-
     # check explanatory variates' dimensions
     if tested_vars.ndim == 1:
         tested_vars = np.atleast_2d(tested_vars).T
 
-    n_samples, n_regressors = tested_vars.shape
+    # check n_sample consistent across arguments
+    if target_vars.shape[0] != tested_vars.shape[0]:
+        raise ValueError(
+            "'target_vars' and 'tested_vars' "
+            "first dimension should be equal.\n"
+            f"Got: {target_vars.shape[0]} for 'target_vars'"
+            f"and {tested_vars.shape[0]} for 'tested_vars'."
+        )
+    if (
+        confounding_vars is not None
+        and target_vars.shape[0] != confounding_vars.shape[0]
+    ):
+        raise ValueError(
+            "'target_vars' and 'confounding_vars' "
+            "first dimension should be equal.\n"
+            f"Got: {target_vars.shape[0]} for 'target_vars'"
+            f"and {confounding_vars.shape[0]} for 'confounding_vars'."
+        )
 
-    # check if explanatory variates contain an intercept (constant) or not
-    intercept_test = False
-    if n_regressors == np.unique(tested_vars).size == 1:
-        intercept_test = True
-
-    # check if confounding vars contains an intercept
-    if confounding_vars is not None:
-        # Search for all constant columns
-        constants = [
-            x
-            for x in range(confounding_vars.shape[1])
-            if np.unique(confounding_vars[:, x]).size == 1
-        ]
-
-        # check if multiple intercepts are defined across all variates
-        if (intercept_test and len(constants) == 1) or len(constants) > 1:
-            # remove all constant columns
-            confounding_vars = np.delete(confounding_vars, constants, axis=1)
-            # warn user if multiple intercepts are found
-            warnings.warn(
-                category=UserWarning,
-                message=(
-                    'Multiple columns across "confounding_vars" and/or '
-                    '"target_vars" are constant. Only one will be used '
-                    "as intercept."
-                ),
-            )
-            model_intercept = True
+    return tested_vars
+
+
+def _check_for_intercept_in_tested_var(tested_vars):
+    """Check if explanatory variates contain an intercept (constant) or not.
+
+    Parameters
+    ----------
+    tested_vars : array-like, shape=(n_samples, n_regressors)
+        Explanatory variates, fitted and tested independently from each others.
+    """
+    n_regressors = tested_vars.shape[1]
+    return n_regressors == np.unique(tested_vars).size == 1
```

Comment:
this checks that there is only one regressor -- so the tested_vars contain _only_ an intercept?

Reply:
and that's really a subjective detail but I see this line and a few others were rewritten based on a linter's suggestions in #3407 and used to be `(a == 1) and (b == 1)`. comparison oparators can be chained in python, but that's not used super frequently and in some other languages it is parsed as 2 binary comparison expressions (so `2 == 2 == 2` is `false`), or is a syntax error -- for example C and C++, Javascript, Octave (and thus I guess Matlab), Java, R, and Rust. so the old version is the "least surprising" one for people relatively new to Python IMO